File: solutions/general/add_subtract_curry/approach1.js
function add_subtract(n) {
  let acc = n;
  let op = true;

  return function inner(m) {
    if (!m) {
      return acc;
    }
    acc += op ? m : -m;
    op = !op;
    return inner;
  };
}

console.log(add_subtract(7)()); // Output: 7
console.log(add_subtract(1)(2)(3)()); // Output: 0
console.log(add_subtract(-5)(10)(3)(9)()); // Output: 11

File: solutions/general/add_subtract_curry/approach1.py
def add_subtract(n):
    acc = n
    op = True
    def inner(m=None):
        nonlocal acc
        nonlocal op
        if m is None:
            return acc
        if op:
            acc += m
        else:
            acc -= m
        op = not op
        return inner
    return inner

# Examples
print(add_subtract(7)())          # Output: 7
print(add_subtract(1)(2)(3)())    # Output: 0
print(add_subtract(-5)(10)(3)(9)()) # Output: 11
